### ISSUE TITLE
Clarification on the representation of a fiber

### ISSUE THREAD

**Issue by vigji** [3d-pli/fastpli#5]
I would like to as for clarification as to why a Fiber is not described in Python as its own class, it would probably be better to validate the input data for generating it and ensuring without further checking that all subsequent functions operating on Fibers, FiberBundles etc get consistent input.

Regardless, I was also a bit confused at first that in the module named objects all the transformations are functions, but capitalised as classes should; is there a reason? I have to say that I am still trying to fully understand the interaction between the C++-compiled part of the code, and the Python parts that make use of such modules; if it is related to that, I would be happy to be given additional pointers on the topic. Probably it would be good if a documentation section could also deal with explaining a bit further the C++/Python integration. If there is such file and I missed it apologies, and please link it to me!

Actually if it is in the user generated documentation, see #4

**Comment by vigji** [comment]
Related to this, reading in the tutorials the representation of FiberBundles properties in a separate list from the list of FiberBundles feels a bit funny (when you define `simpli.fiber_bundles_properties = [[(1.0, -0.001, 10, 'p')]]`). Is there a reason not to keep in a single object a FiberBundle and its properties?

**Comment by fmatuschke** [comment]
I thought about representing the fibers as their own classes. I decided against it because I want to give the user as much freedom as possible to interact with the objects during the building process of models. Additionally, I run into some problems along the pybind11 integration process. But this was already quite some time ago. 
You are correct in suggesting a documentation part about the C++ and Python interaction. Since I am currently in redesigning the Wiki (regarding #4, #5 and #6) I will add such a section. 

The capitalization of the functions was a mistake. I corrected it.

Regarding your question about the `fiber_bundle_properties`. The `simpli` class is split into two processes. The generation and simulation stage. The `run_generation` function generate a discretized 3d representation of the fiber models. Only for the simulation part the properties are necessary. Additionally, an open question is yet which properties should be used for the fibers. Nerve fibers are surrounded by a myelin. The myelin are "lipid layers" wrapped around the axon. How many layers, thickness, different optical properties is not yet fully known. Therefore, the user is capable of changing the properties on the same geometrical model. Additionally, it is possible to run simulations if a discretized volume is provided. That's why I split the properties from the pure geometrical representation. I will try to make this clear inside the wiki and examples.

FYI, this toolbox is capable of building and simulating very large datasets. I tried to make everything as fast as possible. Additionally, I also had a main focus on trying to prevent copping of data. Between Python and C++ I found it quite easy if I am using NumPy arrays. I will try to make this clear inside the Wiki as well.

**Comment by fmatuschke** [comment]
I thought about your comment about the objects, and you were right. I wrote the fiber/fiber_bundle and fiber_bundels objects into their own class. All the previous methods are now class functions of the individual objects. In addition, a fiber_bundle has a `layer` property that represents the properties previously set separately.
I tried to explain the functionality within my new wiki and tutorials. Each method is additionally listed in the API documentation.
I hope this makes it easier for users to get started with my software.
